osu.Game/OsuGame.cs

```diff
@@ -765,11 +767,17 @@ private void forwardLoggedErrorsToNotifications()
 
         private Task asyncLoadStream;
 
+        /// <summary>
+        /// Schedules loading the provided <paramref name="d"/> in a single file.
+        /// </summary>
+        /// <param name="d">The component to load.</param>
+        /// <param name="add">The method to invoke for adding the component.</param>
+        /// <param name="cache">Whether to cache the component as type <typeparamref name="T"/> into the game dependencies before any scheduling.</param>
         private T loadComponentSingleFile<T>(T d, Action<T> add, bool cache = false)
             where T : Drawable
         {
             if (cache)
-                dependencies.Cache(d);
+                dependencies.CacheAs(d);
```

Comment:
I find this a little dangerous. I assume you've changed the `Cache` to `CacheAs` so that components aren't cached with their most derived type (returned at runtime), but with their *declaration* type as written in the field/local. This is an extremely subtle change that can cause regressions if any of the types anywhere don't match - did you make sure that *none* of the single-file-loaded cached dependencies break?

Reply:
Yes, everything is working.

Reply:
If I may ask, how did you determine that? To be able to say that with certainty would require to inspect every BDL/`[Resolved]` member that uses the types cached here and make sure none of them use a more-derived type than the one declared in this class.

Reply:
That's the same as checking for every `loadComponentSingleFile<T>` call to look for ones that have the instantiated type more-derived than the `T` generic type, in which case I couldn't find one ([other than the already-resolved case in the branch](https://github.com/ppy/osu/pull/8965/commits/c025814f403dc5fcf4f07342791387986c526809#diff-4af0c24c8977f999b5986c46cc749ec3R93)), that would mean a confirmed yes.

Moreover, caching components as `T` type IMO is better than most-derived type for this case of loading game components, just like what's being done here (with `CreateUpdateManager()` instance cached as `UpdateManager`), there may be specific components with platform-specific subclasses that need to be cached as their one shared base class to allow resolving them throughout the game.

Reply:
I don't necessarily disagree with the content of the change, I mostly wanted to make sure it was tested as safe.